kolibri/core/discovery/utils/network/client.py

```diff
@@ -115,13 +115,13 @@ def post(self, path, **kwargs):
 
     def request(self, method, path, **kwargs):
         response = None
-        if "timeout" not in kwargs:
-            kwargs.update(timeout=self.timeout)
+
+        timeout = kwargs.pop("timeout", self.timeout)
```

Comment:
@bjester I saw the comment in the filters about wanting to pass the params to the API directly and started investigating. I was running into an error when I tried to pass the filter values as kwargs and discovered that the parent request method expects the params kwarg.

Looking at changing the front-end code to pass these values in to the API seemed like maybe out of scope for this PR, but does this look right and like it might help simplify the facility filters a bit. I was thinking I might just put that into the `apiParams` object created in SelectDeviceForm but paused to get your thoughts here.

Reply:
Yeah so I think the changes that are needed are external to this client class. This class is a generic client utility. This means we rather need to update the APIs that are using it to pass the params.